@xen-orchestra/vhd-lib/src/vhd.js

```diff
@@ -45,28 +53,37 @@ export const PLATFORM_W2KU = 0x57326b75
 export const PLATFORM_MAC = 0x4d616320
 export const PLATFORM_MACX = 0x4d616358
 
-// Other.
-const BLOCK_UNUSED = 0xffffffff
-const BIT_MASK = 0x80
+const footerCookie = 'conectix'
+const creatorApp = 'xo  '
+// it looks like everybody is using Wi2k
+const WIN2K_OS = 0x5769326b
 
-// unused block as buffer containing a uint32BE
-const BUF_BLOCK_UNUSED = Buffer.allocUnsafe(VHD_ENTRY_SIZE)
-BUF_BLOCK_UNUSED.writeUInt32BE(BLOCK_UNUSED, 0)
-
-// ===================================================================
+export const VHD_PARENT_LOCATOR_ENTRIES = 8
 
 const SIZE_OF_32_BITS = Math.pow(2, 32)
-const uint64 = fu.derive(
+
+export const uint64 = fu.derive(
   fu.uint32(2),
   number => [Math.floor(number / SIZE_OF_32_BITS), number % SIZE_OF_32_BITS],
   _ => _[0] * SIZE_OF_32_BITS + _[1]
 )
+export const uint64Undefinable = fu.derive(
+  fu.uint32(2),
+  number =>
+    number === undefined
+      ? [0xffffffff, 0xffffffff]
+      : [Math.floor(number / SIZE_OF_32_BITS), number % SIZE_OF_32_BITS],
+  _ =>
+    _[0] === 0xffffffff && _[1] === 0xffffffff
+      ? undefined
+      : _[0] * SIZE_OF_32_BITS + _[1]
+)
 
-const fuFooter = fu.struct([
+export const fuFooter = fu.struct([
   fu.char('cookie', 8), // 0
   fu.uint32('features'), // 8
   fu.uint32('fileFormatVersion'), // 12
-  uint64('dataOffset'), // offset of the header, should always be 512
+  uint64Undefinable('dataOffset'), // offset of the header
```

Comment:
Can you explain this change?

Reply:
Should we really take care of it?

I think we should explicitly only support dynamic and differencing disks.

Reply:
VMDK to VHD relies on fixed VHDs.